src/args.c

```diff
@@ -24,9 +24,20 @@ args_init(int init_argc, char **init_args)
 	args->argc = 0;
 	args->argv = (char **)x_malloc(sizeof(char *));
 	args->argv[0] = NULL;
-	for (int i = 0; i < init_argc; i++) {
-		args_add(args, init_args[i]);
-	}
+
+    for (int i = 0; i < init_argc; i++) {
+        if(strstr(init_args[i], "-MF") == NULL) {
```

Comment:
This is not the right place for parsing specific options – the `args_*` functions are generic and should stay that way. Please move the logic to the place in `cc_process_args` where `-MF` is handled.